test/assertions_test.rb

```diff
@@ -316,40 +316,57 @@ def test_nested_assertions
     end
   end
 
-  def test_assertion_with_exceptions
+  def test_assertion_block_with_expected_exceptions
     assert_no_assertion_triggered do
-      @test_case.assert_raises(RuntimeError) do
-        @test_case.assert_statsd_increment('counter') do
-          StatsD.increment('counter')
-          raise "foo"
+      @test_case.assert_statsd_increment('expected_happened') do
+        @test_case.assert_raises(RuntimeError) do
+          begin
+            raise "expected"
+          rescue
+            StatsD.increment('expected_happened')
+            raise
+          end
         end
       end
     end
 
-    assert_no_assertion_triggered do
+    assertion = assert_assertion_triggered do
```

Comment:
These helpers seem overly complicated.  Seems like this could just be `exc = assert_raises(Minitest::Assertion) do` and the `assert_no_assertion_triggered` method could just be omitted, since the exception it is catching would fail the test if not caught.